packages/next/src/server/app-render/app-render.tsx

```diff
@@ -987,7 +987,10 @@ async function renderToHTMLOrFlightImpl(
         }
 
         return { stream }
-      } catch (err) {
+      } catch (caughtError: any) {
+        const originalError = digestErrorsMap.get(caughtError.digest)
```

Comment:
We probably want to track RSC errors in a separate bucket from SSR errors. Doing this sort of error replacement really only makes sense if we have an error thrown by the SSR render and we want to map it back to a real error in the RSC runtime. The way it stands we'll always map the thrown error to something but that something may just be a circular reference to the thrown error. This is "correct" but it doesn't really model the semantics of the code which is

"If SSR throws but it matches an error produced by RSC let's use the RSC error instead of the SSR error because it has more information"